api/src/main/java/org/pmiops/workbench/ras/RasLinkService.java

```diff
@@ -119,62 +121,95 @@ public class RasLinkService {
   private final IdentityVerificationService identityVerificationService;
   private final Provider<OpenIdConnectClient> rasOidcClientProvider;
 
+  private final Provider<DbUser> userProvider;
+
   @Autowired
   public RasLinkService(
       AccessModuleService accessModuleService,
       UserService userService,
       IdentityVerificationService identityVerificationService,
-      @Qualifier(RAS_OIDC_CLIENT) Provider<OpenIdConnectClient> rasOidcClientProvider) {
+      @Qualifier(RAS_OIDC_CLIENT) Provider<OpenIdConnectClient> rasOidcClientProvider,
+      Provider<DbUser> userProvider) {
     this.accessModuleService = accessModuleService;
     this.userService = userService;
     this.identityVerificationService = identityVerificationService;
     this.rasOidcClientProvider = rasOidcClientProvider;
+    this.userProvider = userProvider;
   }
 
   /** Links RAS account with AoU account. */
   public DbUser linkRasAccount(String authCode, String redirectUrl) {
     OpenIdConnectClient rasOidcClient = rasOidcClientProvider.get();
     JsonNode userInfoResponse;
+    String txnClaim;
+    String aouUsername = userProvider.get().getUsername();
     try {
       // Oauth dance to get id token and access token.
       TokenResponse tokenResponse =
           rasOidcClient.codeExchange(authCode, decodeUrl(redirectUrl), RAS_AUTH_CODE_SCOPES);
 
+      // The txn claim is used by the RAS team in order to associate our logs with theirs.
+      txnClaim =
+          decodedJwt(tokenResponse.get(ACCESS_TOKEN_FIELD_NAME).toString())
+              .getClaim(TXN_CLAIM)
+              .asString();
+
+      log.info(
+          String.format(
+              "User (%s) received a valid RAS access token, txn:  %s", aouUsername, txnClaim));
       // Validate IAL status.
       String acrClaim =
           decodedJwt(tokenResponse.get(Id_TOKEN_FIELD_NAME).toString())
               .getClaim(ACR_CLAIM)
               .asString();
 
       if (!isIal2(acrClaim)) {
-        log.warning(String.format("User does not have IAL2 enabled, acrClaim: %s", acrClaim));
+        log.warning(
+            String.format(
+                "User does not have IAL2 enabled, acrClaim: %s, txn: %s", acrClaim, txnClaim));
         throw new ForbiddenException(
             String.format("User does not have IAL2 enabled, acrClaim: %s", acrClaim));
```

Comment:
the exceptions in this file don't have the `txn` - would this be useful? @evrii

Reply:
That's a good question. Do you know, off hand how exceptions are handled in our logging? If Exceptions are always logged, then I might just be able to add txn to the exception and get rid of the log statement.

Reply:
not sure